app/lib/widgets/wallets/add_edit_contact.dart

```diff
@@ -81,31 +83,34 @@ class _AddEditContactState extends State<AddEditContact> {
 
   bool _validateAddress(String contactAddress, {bool edit = false}) {
     addressError = null;
+    final allContacts = widget.getAllContacts != null
+        ? widget.getAllContacts!()
+        : widget.contacts;
 
     if (contactAddress.isEmpty) {
       addressError = "Address can't be empty";
       return false;
     }
-    final contacts = widget.contacts.where((c) => c.address == contactAddress);
-    if (edit && contactAddress != widget.address && contacts.isNotEmpty) {
-      addressError = 'Address is used in another contact';
-      return false;
-    } else if (!edit && contacts.isNotEmpty) {
-      addressError =
-          'Address exists in another ${contacts.first.type.name} contact';
 
+    final isDuplicate = allContacts.any((c) => c.address == contactAddress);
+
+    if (isDuplicate) {

```

Comment:
What if we are in the edit flow, and the address is not changed? it will be misleading

Reply:
should be fixed.